ste/mgr-JobPartMgr.go

```diff
@@ -844,6 +845,12 @@ func (jpm *jobPartMgr) Close() {
 	jpm.httpHeaders = common.ResourceHTTPHeaders{}
 	jpm.metadata = common.Metadata{}
 	jpm.preserveLastModifiedTime = false
+
+	/*
+	 * Setting this pipeline nil as it breaks circular dependency, where jpm not freed as it has reference
+	 * in pipeline and pipeline not freed as jpm has pipeline reference.
+	 */
```

Comment:
nit, let's again write a TODO style comment as above.
Avoid using the circular dependency term as that's not actually correct. Even if there's a circular dependency, what matter is reachability from stack or global variables, so if jpm loses references from all those places, it'll get GC'ed even if the pipeline (which itself is not referenced from any stack/global variables) references it.